src/screens/ExchangeModal.js

```diff
@@ -332,19 +364,136 @@ class ExchangeModal extends Component {
       gasUpdateTxFee();
       return this.setState({
         approvalCreationTimestamp: null,
-        approvalEstimatedTimeInMs: null,
+        estimatedApprovalTimeInMs: null,
         isAssetApproved,
         isUnlockingAsset: false,
       });
     }
   };
 
-  getMarketDetails = async isNewOutputReserveCurrency => {
+  updateTradeExecutionDetails = () => {
+    const { chainId, inputReserve, nativeCurrency, outputReserve } = this.props;
+    const {
+      inputAmount: _inputAmount,
+      inputCurrency,
+      inputAsExactAmount: _inputAsExactAmount,
+      outputAmount,
+      outputCurrency,
+    } = this.state;
+
+    const { address: inputAddress, decimals: inputDecimals } = inputCurrency;
+    const { address: outputAddress, decimals: outputDecimals } = outputCurrency;
+
+    const isInputEth = inputAddress === 'eth';
+    const isOutputEth = outputAddress === 'eth';
+
+    let inputAmount = _inputAmount;
+    let inputAsExactAmount = _inputAsExactAmount;
+    const isMissingAmounts = !inputAmount && !outputAmount;
+
+    if (isMissingAmounts) {
+      const DEFAULT_NATIVE_INPUT_AMOUNT = 50;
```

Comment:
I guess there's not much point on declaring this const other than readability.

Reply:
yeah, at least we can move it up and out of this function. Actually, I see a bit of cleanup we can do here in this file (but maybe we should wait until we make the savings/exchange modal modular)